### ISSUE TITLE
False positive with unquote_splicing

### ISSUE THREAD

**Issue by hissssst** [rrrene/credo#972]
### Environment

```
credo 1.6.4
```

### What were you trying to do?

```elixir
quote do
  with unquote_splicing(cases) do
    {:ok, unquote(ret)}
  else
    _ -> :error
  end
end
```

### Expected outcome

No warnings

### Actual outcome
```
┃ [R] → `with` contains only one <- clause and an `else` branch, consider using `case` instead
```

**Comment by rrrene** [comment]
Why do you think this should not raise an issue?

Your code is equivalent to

```elixir
  if unquote_splicing(cases) do
    {:ok, unquote(ret)}
  else
    _ -> :error
  end
```

which Credo points out.

You don't have to agree with this assesment, but I think this is not a bug unless I am missunderstanding you (note that you did not provide any explanation in this issue).

**Comment by hissssst** [comment]
Hi, that is a good question. You can check the documentation for `unquote_splicing` inside the `quote` [here](https://hexdocs.pm/elixir/Kernel.SpecialForms.html#unquote_splicing/1). Basically, this means that there can be multiple clauses (one or more), so this warning is not correct (because this warning assumes that there's only one clause)

**Comment by rrrene** [comment]
Ah, my mistake. You are right 👍
